### PR TITLE
Late Patches

### PR DESCRIPTION
**Pull request by Plypy**
This PR contains 4 patches wrapped in three independent commits.

As the commit messages suggested, they are meant to,
- Used `uuid` to mark the error that would happen, and only show the stack trace in development. ID-85 complained this.
- Add back the missing method, `ldap.addGroup` and changed a `this` to `that` in `models/group.js
- Make dashboard remember what options user chose for welcome message.
- Support user to login via their email, any email not only primary one. Also fixed validation error when user posted email in their username column.

### CODE REVIEW

**Comment by Plypy** [comment]
I'll make some patches for production branch later.

Specifically, error tracking and email login.

File: app/models/group.js [hunk 1]
```diff
@@ -59,12 +59,13 @@ groupSchema.pre('save', function (next) {
   if (this.inLDAP) {
     return next();
   }
+  var that = this;
```
Comment (robbyoconnor) [inline app/models/group.js:62]: why do you have this -- I **REALLLLLLLLY** hate this -- it's very unclear There has to be a cleaner way

Comment (robbyoconnor) [inline app/models/group.js:62]: It's not enough to stop me from merging it -- I'll probably fix it at some point but it's not an issue for now...just wanted to have it on record :smile_cat:

**Comment by robbyoconnor** [comment]
@Plypy -- don't make a PR against production, going to deploy to staging within the coming weeks and maybe cut a release -- I believe we've hit all 2.1 milestones 

@elliottwilliams  -- you on board with cutting a release after this?

**Comment by robbyoconnor** [comment]
I also need to go on record two of these commits don't count as GSoC -- keep this PR as-is; the merge commit msg will reflect this. 

The issue is that they are done beyond the Aug 21 Pencil's Down date. Do not include them in your final code submission on Melange. Great work! 

Two that count are: 
- 903839f1f29861353a2606c36a8a299b419d6daa
- 5f2c8377e6fb21d4b7bd4e589b8c926cdd08f35e

Two that do not are: 
- 886b63871f6e850fd6959b9033c706f95f90a010
- 0bf362c18d897847f42088ba54d63b1493261623

**Pull request merged**